### DISCUSSION TITLE
Ubuntu LTS Personal Package Archives and ImageMagick 7

### DISCUSSION THREAD

**Discussion by boan-jfm** [ImageMagick/ImageMagick#6612] [General] [answered]
Hello

Why isn't ImageMagick 7 included in the Ubuntu (2204) PPA? As far as I know, ImageMagick 7 has been available since 2016, and I'm curious why Ubuntu hasn't updated to this version.

Thank you.

**Comment by dlemstra** [comment, marked as answer]
We don't know? We only provide the library and we have no control over which version is used in a distribution. You should probably ask this question somewhere else.

**Reply by boan-jfm** [reply]
I did some more research and found this: https://askubuntu.com/a/1201973

> Check out the Debian Package Tracker for Imagemagick to see some of the issues.
>
> * The automated version monitoring tool does not yet (as of today) register 7.0 as being available. So nobody at Debian knows to package it.
>
> * Take a look at the many HIGH and NORMAL issues in the "Action Needed" section. A lot of volunteer effort is needed before another version can be packaged at all!
>
> So I would evaluate the chances of Imagemagick 7 making it into Ubuntu 20.04 as LOW.
>
> It's possible if a group of Debian-skilled volunteers charge in and help the current maintainers work all those problems and package 7.0.9 immediately...it's just not likely that such a group is about to do that.

**Reply by boan-jfm** [reply]
I asked in the Debian/Ubuntu community and got an answer from [Raphaël Hertzog](https://salsa.debian.org/hertzog), a maintainer on the [Debian distro tracker](https://salsa.debian.org/qa/distro-tracker/-/commits/master). He referenced a [bug](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=929825) and suggested to poke the maintainers for their plan?

>Hi,
>
>A lot:
>- command line API is not fully compatible with v6
>- I am trying to push latest v6, but it is blocked in NEWS and need a transition
>
>I plan imagemagick7 to sit in experimental in parallel with
>imagemagick6 at least before stabilazing
>
>Could you help to triage bug, and fix security issue in v6 ? It will
>help me and allow me to concentrate in v6->v7 transition
>
>Bastien
> -- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=929825

**Comment by rich2005** [comment]
Many other packages have IM6 as a dependency, having IM7 as a default package breaks them.

In the past I would compile IM7 as a static package and run that in parallel with IM6.

At the moment the linux IM7 appimage works fine, also runs in parallel with IM6.

Using (k)ubuntu 20.04

**Reply by boan-jfm** [reply]
ImageMagick 7 was released back in 2016, and it might be time to consider including it as part of the PPA so that users have the option to update.

![image](https://github.com/ImageMagick/ImageMagick/assets/13201214/ea5a38b7-e651-487f-9dbe-46bdf22faa08)
https://imagemagick.org/archive/releases/

For security reasons, our organization relies on maintained software from the PPA and avoids manual installations and updates.

Are there any issues having both in the PPA, with version 6 as default?

**Reply by rich2005** [reply]
_For security reasons, our organization relies on maintained software from the PPA and avoids manual installations and updates._

That is a bit of a constraint, IM7 is very easy to compile and the magick appimage is a single file to download / install.

The problem with PPA's is the personal bit, they are maintained by individuals not ubuntu. Maybe find someone who is willing to add IM7 to their PPA, plenty to choose from https://launchpad.net/ubuntu/+ppas?name_filter=imagemagick

..or start one yourself. Nice if the IM team created one but they already provide debs and rpms for a variety of distros, maybe they do not want the added work.

edit: A quick look around and maybe IM7 is in Debian 11 so it might come with the LTS that follows ubuntu 22.10

**Reply by boan-jfm** [reply]
>edit: A quick look around and maybe IM7 is in Debian 11 so it might come with the LTS that follows ubuntu 22.10

Can you send a reference to this? 

Debian 11:
https://packages.debian.org/bullseye/libs/

Debian 12: 
https://packages.debian.org/bookworm/libs/

When I do a library search, it is still ImageMagick 6.9.11.60, I expect it must be part of the tracker before it will be integrated: https://tracker.debian.org/pkg/imagemagick

**Reply by rich2005** [reply]
Maybe this one, why not drop the maintainer an email and ask for advice. It might appear in Debian 11 sometime and migrate to Ubuntu 

https://debian.pkgs.org/11/multimedia-main-amd64/imagemagick-7-common_7.1.0.4-dmo1_all.deb.html

**Reply by sbrl** [reply]
Debian 13 ships with ImageMagick 7.1: https://tracker.debian.org/pkg/imagemagick
